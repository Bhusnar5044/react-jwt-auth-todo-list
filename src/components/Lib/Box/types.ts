export type Props = {
    display?: 'flex' | 'inline-flex';
    flexDirection?: 'row' | 'column';
    justifyContent?: 'center' | 'space-between' | 'space-around';
    alignItems?: 'center' | 'start' | 'end';
    flexWrap?: 'wrap';
    gap?: string;
    margin?: string;
};
