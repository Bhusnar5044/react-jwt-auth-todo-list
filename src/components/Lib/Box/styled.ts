import { styled } from '@utils';
import { Props } from './types';

export const Container = styled.div<Props>`
    display: ${({ display }) => display || 'flex'};
    flex-direction: ${({ flexDirection }) => flexDirection};
    justify-content: ${({ justifyContent }) => justifyContent};
    align-items: ${({ alignItems }) => alignItems};
    flex-wrap: ${({ flexWrap }) => flexWrap};
    gap: ${({ gap }) => gap};
    margin: ${({ margin }) => margin};
`;
