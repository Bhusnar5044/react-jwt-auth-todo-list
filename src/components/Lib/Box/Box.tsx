import { FC, PropsWithChildren, memo } from 'react';
import { Container } from './styled';
import { Props } from './types';

export const Box: FC<PropsWithChildren & Props> = memo(({ children, ...props }) => {
    return <Container {...props}>{children}</Container>;
});

Box.displayName = 'Box';
