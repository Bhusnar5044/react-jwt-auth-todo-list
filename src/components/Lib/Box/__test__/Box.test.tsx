import { render, screen } from '@test-utils';
import { Box } from '../Box';

describe('Box', () => {
    it('should render properly', () => {
        const { container } = render(<Box />);
        expect(container).toMatchSnapshot();
    });

    it('should render with content', () => {
        render(<Box>dummy content</Box>);
        expect(screen.getByText(/dummy content/i)).toBeInTheDocument();
    });

    it('should render with given style', () => {
        render(
            <Box display="flex" flexDirection="column" justifyContent="center" alignItems="center">
                dummy content
            </Box>,
        );
        expect(screen.getByText(/dummy content/i)).toHaveStyleRule(
            'display',
            'flex',
            'flex-direction',
            'column',
            'align-items',
            'center',
        );
    });
});
