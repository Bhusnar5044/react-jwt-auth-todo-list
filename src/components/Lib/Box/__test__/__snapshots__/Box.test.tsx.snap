// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render properly 1`] = `
.c0 {
  display: flex;
}

<div>
  <div
    class="c0"
  />
</div>
`;
